### PR TITLE
Update Sample App Readme

### PR DESCRIPTION
**Pull request by cacheung**
Update Sample App Readme

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by stcheng-adobe — APPROVED** [review]

File: apps/AEPSampleApp/README.md [hunk 1]
```diff
@@ -1,32 +1,66 @@
 # AEPSampleApp
 
-> Note: Commands are assuming you're in the root directory of the repository.
+## Prerequisites
 
-> Note: If you are new to React Native we suggest you follow the [React Native Getting Started](<https://facebook.github.io/react-native/docs/getting-started.html>) page before continuing.
+ * If you are new to React Native we suggest you follow the [React Native Setting up development environment](<https://reactnative.dev/docs/environment-setup>) React Native CLI Quickstart page before continuing.
+ * iOS pod version >= 1.11.3
```
Comment (emdobrin) [inline apps/AEPSampleApp/README.md:6]: Cocoapods >= 1.11.3

File: apps/AEPSampleApp/README.md [hunk 2]
```diff
@@ -1,32 +1,66 @@
...
+## How to run the sample app:
 
 ### Add your App Id:
+
 In `ios/**/AppDelegate.mm`, find the call to `configureWithAppId` and add your app id.
 
 In `android/**/MainApplication.java`, find the call to `configureWithAppId` and add your app id.
 
-#### Install dependecies 
+> Note: App id is configured in a mobile propery in Data Collection UI.  Refer to [configure with App ID per environment](https://developer.adobe.com/client-sdks/home/base/mobile-core/configuration/#configure-with-app-id-per-environment) page for more information.
```
Comment (emdobrin) [inline apps/AEPSampleApp/README.md:16]: - typo: mobile property 
- in Data Collection UI, in Tags.

File: apps/AEPSampleApp/README.md [hunk 3]
```diff
@@ -1,32 +1,66 @@
...
-#### Run instructions for iOS:
+### Run instructions for iOS:
 
 ```
 yarn sampleapp:ios:run
 ```
+Alternatively, running from Xcode: <br>
+In terminal > run `npx react-native start` <br>
+Double click `AEPSampleApp.xcworkspace` file. <br>
+Run the app in Xcode.
```
Comment (emdobrin) [inline apps/AEPSampleApp/README.md:33]: How about we transform these into command line steps:
Alternatively, you can run the iOS app from Xcode by executing the following command in your terminal:
```
npx react-native start
```
Then navigate to apps/AEPSampleApp/ios and double click on the `AEPSampleApp.xcworkspace` file, or from the terminal:
```
cd apps/AEPSampleApp/ios
open AEPSampleApp.xcworkspace
```

File: apps/AEPSampleApp/README.md [hunk 4]
```diff
@@ -1,32 +1,66 @@
...
 
 Have an Android emulator running (quickest way to get started), or a device connected. https://developer.android.com/studio/run/emulator-commandline
 
 ```
 yarn sampleapp:android:run
 ```
+Alternatively, running from Android Studio: <br>
+In terminal > run `npx react-native start` <br>
+Double click `build.gradle` file. <br>
+Run the app in the Android Studio.
```
Comment (emdobrin) [inline apps/AEPSampleApp/README.md:45]: Alternatively, you can run the Android app from Android Studio by executing the following command in your terminal:
```
npx react-native start
```

Then navigate to apps/AEPSampleApp/android, double click on the `build.gradle` file, and run the app in Android Studio.

File: apps/AEPSampleApp/README.md [hunk 5]
```diff
@@ -1,32 +1,66 @@
...
+
+## Resources and Troubleshooting
+* Where can we find the information about the Yarn scripts in the Sample App? <br>
+  Refer to [Sample App yarn scripts](https://github.com/adobe/aepsdk-react-native/blob/main/package.json#L11).
+
+* Where is the sample app React Native code? <br>
+  All the extension TypeScript code is located under AEPSampleApp > extensions
+
+* I encount errors when building the app. <br>
```
Comment (emdobrin) [inline apps/AEPSampleApp/README.md:54]: I encountered errors while building the app.

File: apps/AEPSampleApp/README.md [hunk 6]
```diff
@@ -1,32 +1,66 @@
...
+  Validate you have node and nvm installed, check your React Native development setup, refer to the doc under Prerequisites.
```
Comment (emdobrin) [inline apps/AEPSampleApp/README.md:55]: let's include the nvm and node min version in the prerequisites

File: apps/AEPSampleApp/README.md [hunk 7]
```diff
@@ -1,32 +1,79 @@
...
+ * Node version >= 18
 
-How to run the sample app:
+## How to run the sample app:
 
 ### Add your App Id:
+
 In `ios/**/AppDelegate.mm`, find the call to `configureWithAppId` and add your app id.
 
 In `android/**/MainApplication.java`, find the call to `configureWithAppId` and add your app id.
```
Comment (emdobrin) [inline apps/AEPSampleApp/README.md:16]: - can we use the full path for these files from the root directory, e.g. apps/AEPSampleApp/ios/AEPSampleApp/AppDelegate.mm
- can we add simple instructions on where to find the app id?

Comment (cacheung) [reply apps/AEPSampleApp/README.md:16]: I have a note section right below to point to the user how to setup mobile property.  Is that good enough?  We should assume them already have the knowledge to setup the app id and schema, etc before going to this sample app.

App id is configured in a mobile property in Data Collection UI, in Tags. Refer to [configure with App ID per environment](https://developer.adobe.com/client-sdks/home/base/mobile-core/configuration/#configure-with-app-id-per-environment) page for more information.

**Review by dsoffiantini — APPROVED** [review]

**Pull request merged**